web/src/compose_actions.ts

```diff
@@ -339,6 +339,7 @@ export let start = (raw_opts: ComposeActionsStartOpts): void => {
         compose_recipient.on_compose_select_recipient_update();
     } else if (opts.stream_id && opts.topic) {
         compose_state.set_stream_id(opts.stream_id);
+        compose_state.topic(opts.topic);
```

Comment:
This is generally being set after this big if/else block for all cases, see the call to `compose_recipient.update_topic_displayed_text(opts.topic)` below.

So this just sets the topic for this case a little bit earlier, so that the helper functions that are called via `compose_recipient.on_compose_select_recipient_update()` have the full channel/topic pair in this case.